### ISSUE TITLE
Decouple Responses and Domain

### ISSUE THREAD

**Issue by mjspeck** [RasaHQ/rasa#5307]
**Description of Problem**:

Currently responses live in `domain.yml`, where intents, actions, slots, and the session config also live. While it's nice to have everything in one place, it's unnecessary and difficult for rapid development. Since editing the domain requires retraining, this means that editing a response forces retraining the model, even if none of the actions, intents, slots, or session config are different. This is unnecessary because the core model is still making the exact same prediction. 

**Overview of the Solution**:

Editing the code in a custom action doesn't cause model retraining, why should editing a response? I'm suggesting separating the responses out into a separate file like `responses.yml`, or something similar. Whenever an `utter_` action is called, it simply references this file.

**Examples** (if relevant):

**Blockers** (if relevant):

**Definition of Done**:

**Comment by amn41** [comment]
hi @mjspeck - thanks for the suggestion! We have thought about this, but the nice thing about coupling model training and nlg is that you can guarantee your assistant will behave exactly the same way if you run a model again at a later time. 

fwiw decoupling this is independent of separating into different files. We can (and should) have `rasa train` notice when only responses have changed and just create a new model tarball without actually training any new models.

**Comment by mjspeck** [comment]
Yeah that makes sense to me. And that functionality would be greatly appreciated.

**Comment by akelad** [comment]
@amn41 @mjspeck actually this has already been the case for quite some time. If you train a model and only the responses change it looks like this:
```
(rasa) akela@BERMB00007 rasa-test % rasa train
2020-03-19 13:58:03 INFO     rasa.model  - Data (nlg) for NLG templates section changed.
Core stories/configuration did not change. Only the templates section has been changed. A new model with the updated templates will be created.
NLU data/configuration did not change. No need to retrain NLU model.
```

**Comment by koaning** [comment]
I've noticed during workshops that if you'd like to see the effect of changing a response in the  `domain.yml` file it can be a bit odd to have to run this;

```
rasa train --force; rasa shell
```

in order to see the effect. It's not a huge blocker, but it is a caveat.

**Comment by akelad** [comment]
that shouldn't be the case, if you're seeing that then that's a bug - what version are you on?

**Comment by wochinge** [comment]
Fixed with 2.0 training data.
